### ISSUE TITLE
EncryptionEventListener not initialised if `mongodb.encrypt.silent-decryption-failures` is not configured

### ISSUE THREAD

**Issue by rainerfrey-inxmail** [bolcom/spring-data-mongodb-encrypt#48]
Our application is configured with encryption type `reflection`, and `silent-encryption-failures` left to the default (false).

```yaml
mongodb.encrypt:
  type: reflection
  keys:
    - version: 1
      key: ...
```
After upgrading to 2.7.0, I get the following exception:

```
No converter found capable of converting from type [org.bson.types.Binary] to type [java.lang.String]
org.springframework.core.convert.ConverterNotFoundException: No converter found capable of converting from type [org.bson.types.Binary] to type [java.lang.String]
```

Debug output shows in the condition evaluation report under negative matches:

```
   EncryptAutoConfiguration#encryptionEventListener:
      Did not match:
         - @ConditionalOnProperty (mongodb.encrypt.[type,silent-decryption-failures]) did not find property 'silent-decryption-failures' (OnPropertyCondition)
```

The culprit seems to be this line in the auto configuration:
https://github.com/bolcom/spring-data-mongodb-encrypt/blob/44b9f247191b400316eb8eff13d21189c83f7602/src/main/java/com/bol/config/EncryptAutoConfiguration.java#L30

Apparently `@ConditionalOnProperty` (with no `matchIfMissing`) only evaluates to true if both properties `type` *and* `silent-encryption-failures` are explicitly configured. I guess that the intention was that one of the properties need to be set.

**Comment by rainerfrey-inxmail** [comment]
This also occurs if `mongodb.encrypt.silent-decryption-failures` is explicitly set to `false`.

**Comment by agoston** [comment]
You are right! The idea was that autoconfiguration should only happen if there are any properties set under `mongodb.encrypt`, but there is just no such option that I could find.

Documentation of `@ConditionalOnProperty` is really scarce regarding if list elements are considered in and AND or an OR relationship. And in system test, both are set for testing purposes.

Let me check if I can find a solution to this. Until then, a workaround is to explicitly set `mongodb.encrypt.[type,silent-decryption-failures]`.

**Comment by rainerfrey-inxmail** [comment]
> You are right! The idea was that autoconfiguration should only happen if there are any properties set under `mongodb.encrypt`, but there is just no such option that I could find.
> 
> Documentation of `@ConditionalOnProperty` is really scarce regarding if list elements are considered in and AND or an OR relationship.

I wouldn't habe know that behaviour either. I think Spring's own code just uses custom condition implementations (mostly as local inner classes in the autoconfiguration class) in such cases.

**Comment by agoston** [comment]
Correct, that's what I noticed in other projects too. But I really want to keep this library simple and short, so custom wiring annotations falls well outside the scope.

Now I've added a simple change, along with a system test to prove it working. Take a peek if interested.

Also released 2.7.1 with this change.
